### PR TITLE
Fix bug with finding a full moon

### PR DESCRIPTION
**Pull request by damon140**
Fixes a bug where if the given date is very close to a full moon then the wrong full moon was returned.

### CODE REVIEW

File: src/main/java/com/bradsbrain/simpleastronomy/MoonPhaseFinder.java [hunk 1]
```diff
@@ -36,81 +38,119 @@
...
+	 */
+	private static ZonedDateTime _findFirstAnswerAfter(ZonedDateTime cal, MoonFinder finder) {
+		// To account for rounding problems, make several dates near
+		// and choose the best answer.
+		ZonedDateTime nearDatePast = cal.minusDays(1);
+    	ZonedDateTime nearDateFuture = cal.plusDays(1);
+    	
+    	ZonedDateTime answer1 = _findRoundedDatePassingBounds(nearDatePast, finder);
+    	ZonedDateTime answer2 = _findRoundedDatePassingBounds(cal, finder);
+		ZonedDateTime answer3 = _findRoundedDatePassingBounds(nearDateFuture, finder);
```
Comment (dustmachine) [inline src/main/java/com/bradsbrain/simpleastronomy/MoonPhaseFinder.java]: minor thing, but please ensure whitespace is always spaces (not tabs)

File: src/main/java/com/bradsbrain/simpleastronomy/MoonPhaseFinder.java [hunk 2]
```diff
@@ -36,81 +38,119 @@
...
+	
+	/**
+	 * Tries several close dates and returns the first answer that is after the input calendar.
+	 * This works around rounding problems that effect the binary search.
+	 * 
+	 * @param cal the date to search from
+	 * @param finder the finder to use
+	 * @return the best answer after cal
+	 */
+	private static ZonedDateTime _findFirstAnswerAfter(ZonedDateTime cal, MoonFinder finder) {
```
Comment (dustmachine) [inline src/main/java/com/bradsbrain/simpleastronomy/MoonPhaseFinder.java]: even though it's private, do not use underscore

**Comment by damon140** [comment]
There is a sort of soak test in FullMoonFinderTest#twentyThousandFinds that checks many points in a month to shake out edge cases.  It found that right before a full moon the binary search method had a problem due to accuracy, which caused it to choose a full moon a month later that it should have.  I fixed this problem by making multiple calculations and choosing the earliest.

**Comment by dustmachine** [comment]
validated that all tests pass.
Thanks for the updates.
